dipy/workflows/align.py

```diff
@@ -1042,7 +1043,7 @@ def run(
             new_tractogram, pjoin(out_dir, out_nonlinear_moved), header=moving_header
         )
 
-        df = pd.DataFrame(warp, columns=["transforms", "gaussian_kernel"])
+        df = pd.DataFrame(warp, columns=["gaussian_kernel", "transforms"])
```

Comment:
Was this in the wrong order before? Why wasn't it catastrophic?

Reply:
Yes, it was in the wrong order, but it was not catastrophic because we use the `df` data frame only to save the warp map and Gaussian kernel. We do not use it to apply any transforms to the bundle itself. Essentially, it was saving NumPy files of transforms and Gaussian kernel with incorrect names (their file names were interchanged).

Reply:
Is there any test we could write to make sure that doesn't creep back in?

Reply:
I made some changes. I moved this to BundleWarp function. Now, registration returns warp as a dataframe with "gaussian_kernel", "transforms" columns. No need to do it in the workflow (outside of the registration function). Also, added test function `test_transformation_dimensions()` that checks if column index and dimensions of transformation matrices are correct. Let me know if there's something else @arokem @skoudoro